File: katas/6 kyu/find-the-parity-outlier/solution.py
# find-the-parity-outlier
import codewars_test as test
# -------------------------------- MY SOLUTION ------------------------------- #
def find_outlier(integers):
  a = [x for x in integers if x % 2 == 0]
  return [x for x in integers if x % 2 != 0 ][0] if len(a) > 1 else a[0]
# ------------------------------ CLEVER SOLUTION ----------------------------- #
def find_outlier(int):
    odds = [x for x in int if x%2!=0]
    evens= [x for x in int if x%2==0]
    return odds[0] if len(odds)<len(evens) else evens[0]
# ----------------------------------- TEST ----------------------------------- #
test.assert_equals(find_outlier([2, 4, 6, 8, 10, 3]), 3)
test.assert_equals(find_outlier([2, 4, 0, 100, 4, 11, 2602, 36]), 11)
test.assert_equals(find_outlier([160, 3, 1719, 19, 11, 13, -21]), 160)

File: katas/6 kyu/find-the-parity-outlier/solution.php
<!-- find-the-parity-outlier -->
<?php
use PHPUnit\Framework\TestCase;
/* ------------------------------- MY SOLUTION ------------------------------ */
function find($integers) {
  $odd = array_filter($integers , function ($x) { return $x%2 !== 0;});
  $even = array_filter($integers , function ($x) { return $x%2 === 0;});
  return count($odd) > count($even) ? end($even) : end($odd);
}
/* ----------------------------- CLEVER SOLUTION ---------------------------- */
function find($i) {
  foreach ($i as $x)
    $x % 2 == 0 ? $even[] = $x : $odd[] = $x;
  return $odd < $even ? $odd[0] : $even[0];
}
/* ---------------------------------- TEST ---------------------------------- */
class Tests extends TestCase
{
  public function testBasic() {
    $this->assertEquals(101, find([100, 101, 102]));
    $this->assertEquals(100, find([1, 9, 27, 81, 100]));
    $this->assertEquals(11, find([2, 4, 0, 100, 4, 11, 2602, 36]));
    $this->assertEquals(160, find([160, 3, 1719, 19, 11, 13, -21]));
  }
}